agent_pool.go

```diff
@@ -0,0 +1,190 @@
+	ListOptions
+}
+
+// List all the agent pools of the given organization. Note this currently is
+// limited to a single pool per organization (API enforced).
+func (s *agentPools) List(ctx context.Context, organization string, options AgentPoolListOptions) (*AgentPoolList, error) {
+	if !validStringID(&organization) {
+		return nil, errors.New("invalid value for organization")
+	}
+
+	u := fmt.Sprintf("organizations/%s/agent-pools", url.QueryEscape(organization))
+	req, err := s.client.newRequest("GET", u, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	poolList := &AgentPoolList{}
+	err = s.client.do(ctx, req, poolList)
+	if err != nil {
+		return nil, err
+	}
+
+	return poolList, nil
+}
+
+// AgentPoolCreateOptions represents the options for creating an agent pool.
+type AgentPoolCreateOptions struct {
+	// For internal use only!
+	ID string `jsonapi:"primary,agent-pools"`
+
+	// A name to identify the agent pool.
+	Name *string `jsonapi:"attr,name"`
+}
+
+func (o AgentPoolCreateOptions) valid() error {
+	if !validString(o.Name) {
+		return errors.New("name is required")
+	}
+	return nil
+}
+
+// Create a new agent pool with the given options. Note only a single pool is
+// allowed per organization (API enforced).
+func (s *agentPools) Create(ctx context.Context, organization string, options AgentPoolCreateOptions) (*AgentPool, error) {
+	if !validStringID(&organization) {
+		return nil, errors.New("invalid value for organization")
+	}
+
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	// Make sure we don't send a user provided ID.
+	options.ID = ""
+
+	u := fmt.Sprintf("organizations/%s/agent-pools", url.QueryEscape(organization))
+	req, err := s.client.newRequest("POST", u, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	pool := &AgentPool{}
+	err = s.client.do(ctx, req, pool)
+	if err != nil {
+		return nil, err
+	}
+
+	return pool, nil
+}
+
+// Read a single agent pool by its ID.
+func (s *agentPools) Read(ctx context.Context, agentpoolID string) (*AgentPool, error) {
+	if !validStringID(&agentpoolID) {
+		return nil, errors.New("invalid value for agent pool ID")
+	}
+
+	u := fmt.Sprintf("agent-pools/%s", url.QueryEscape(agentpoolID))
+	req, err := s.client.newRequest("GET", u, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	pool := &AgentPool{}
+	err = s.client.do(ctx, req, pool)
+	if err != nil {
+		return nil, err
+	}
+
+	return pool, nil
+}
+
+// AgentPoolUpdateOptions represents the options for updating an agent pool.
+type AgentPoolUpdateOptions struct {
+	// For internal use only!
+	ID string `jsonapi:"primary,agent-pools"`
+
+	// A new name to identify the agent pool.
+	Name *string `jsonapi:"attr,name"`
+}
+
+// Update an agent pool by its ID.
+func (s *agentPools) Update(ctx context.Context, agentPoolID string, options AgentPoolUpdateOptions) (*AgentPool, error) {
+	if !validStringID(&agentPoolID) {
+		return nil, errors.New("invalid value for agent pool ID")
+	}
```

Comment:
i think it would be good to validate the name in the options like we do for the create options.